packages/swagger-ts/src/TypeGenerator.ts

```diff
@@ -386,8 +400,47 @@ export class TypeGenerator extends Generator<PluginOptions['resolvedOptions'], C
         })
       }
 
-      if (this.options.dateType === 'date' && ['date', 'date-time'].some((item) => item === schema.format)) {
-        return factory.createTypeReferenceNode(factory.createIdentifier('Date'))
+      /**
+       * > Structural validation alone may be insufficient to allow an application to correctly utilize certain values. The "format"
+       * > annotation keyword is defined to allow schema authors to convey semantic information for a fixed subset of values which are
+       * > accurately described by authoritative resources, be they RFCs or other external specifications.
+       *
+       * In other words: format is more specific than type alone, hence it should override the type value, if possible.
+       *
+       * see also https://json-schema.org/draft/2020-12/draft-bhutton-json-schema-validation-00#rfc.section.7
+       */
+      if (schema.format) {
+        switch (schema.format) {
+          case 'binary':
+            if (schema.type === 'string') {
+              return factory.createTypeReferenceNode('Blob', [])
+            }
+            break
+          // 7.3.1. Dates, Times, and Duration
+          case 'date-time':
+          case 'date':
+          case 'time':
```

Comment:
actually, im not sure if `time` should really be cast a `Date`. Maybe `number` instead?

Reply:
We have `dateType` which can now be a date or a string so maybe we can also add number if that would be needed for other users.